test/PhpInputStreamTest.php

```diff
@@ -64,6 +64,17 @@ public function testGetContentsReturnsRemainingContentsOfStream()
         $this->assertEquals(substr($contents, 128), $remainder);
     }
 
+    public function testGetContentsReturnCacheWhenReachedEof()
+    {
+        $this->stream->getContents();
+        $this->assertStreamContents($this->stream->getContents());
+
+        $stream = new PhpInputStream('data://,0');
```

Comment:
a very trivial data:// URL, without mime, charset or base64. (see RFC 2397, 3. syntax)
In this case, with just `0` as contents.